src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -782,12 +782,14 @@ pImage = do
   let kvs = concatMap getAtt ["width", "height", "sizes", "srcset"]
   return $ B.imageWith (uid, cls, kvs) (escapeURI url) title (B.text alt)
 
-pSamp :: PandocMonad m => TagParser m Inlines 
-pSamp = try $ do 
-  TagOpen open attr' <- pSatisfy $ tagOpen (=="samp") (const True)
+pCodeWithClass :: PandocMonad m => [(T.Text,String)] -> TagParser m Inlines 
+pCodeWithClass elemToClass = try $ do 
+  let tagTest = flip elem . fmap fst $ elemToClass
+  TagOpen open attr' <- pSatisfy $ tagOpen tagTest (const True)
   result <- manyTill pAny (pCloses open)
   let (ids,cs,kvs) = mkAttr . toStringAttr $ attr'
-  return . B.codeWith (ids,"sample":cs,kvs) .
+      cs'          = maybe cs (:cs) . lookup open $ elemToClass
+  return . B.codeWith (ids,cs',kvs) .
     unwords . lines . T.unpack . innerText $ result
```

Comment:
I think this needs to be changed as it won't parse tags inside `<samp>` and `<var>`? Currently it does the following. 

```
➜  pandoc git:(issue5799) ✗pandoc -f html -t markdown
<var><code class=\"sourceCode haskell\"><span class=\"op\">&gt;&gt;=</span></code></var>
`>>=`{.variable}
```

Reply:
That's on purpose.  The `Code` constructor takes a plain String.  So we have to strip out (ignore) all the tags inside it when we're reading.  There's no way to represent them.